src/main/java/org/springframework/data/elasticsearch/repository/query/ElasticsearchStringQuery.java

```diff
@@ -103,4 +120,23 @@ protected StringQuery createQuery(ParametersParameterAccessor parameterAccessor)
 		return new StringQuery(queryString);
 	}
 
+	protected SourceFilter createSourceFilter(ParametersParameterAccessor parameterAccessor) {
+		StringQueryUtil stringQueryUtil = new StringQueryUtil(elasticsearchOperations.getElasticsearchConverter().getConversionService());
+
+		String[] includeList = new String[0];
+		String[] excludeList = new String[0];
+
+		final String sourceFilterDelim = ",";
+		if (includes.length() != 0) {
+			String includeFilter = stringQueryUtil.replacePlaceholders(includes, parameterAccessor);
+			includeList = includeFilter.split(sourceFilterDelim);
+		}
+		if (excludes.length() != 0) {
+			String excludeFilter = stringQueryUtil.replacePlaceholders(excludes, parameterAccessor);
+			excludeList = excludeFilter.split(sourceFilterDelim);
+		}
+
```

Comment:
This splitting is not necessary if the values are already coming as a `String[]` from the annotation.

Reply:
got it